hercules-ci-api/src/Hercules/API/Projects/Job.hs

```diff
@@ -13,17 +13,28 @@ data Job = Job
   , projectId :: Id Project
   , index :: Int64
   , repoId :: Id Repo
-  , revision :: Text
   , startTime :: UTCTime
   , endTime :: Maybe UTCTime
   , jobPhase :: JobPhase
   , jobStatus :: JobStatus
   , evaluationStatus :: JobStatus
   , derivationStatus :: JobStatus
   , evaluationId :: Id Evaluation
+  , source :: GitSource
   }
   deriving (Generic, Show, Eq, ToJSON, FromJSON, ToSchema)
 
+
+data GitSource = GitSource
+  { revision :: Text
+  , ref :: Text
+  , committerName :: Text
+  , committerLogin :: Maybe Text
```

Comment:
What exactly are the name and login?
Some options I can think of:
 - git commit author+email field
 - what github calls a login but we call account [slug](https://github.com/hercules-ci/hercules-ci-agent/blob/master/hercules-ci-api/src/Hercules/API/Accounts/Account.hs#L13) (which should be of type [`Name`](https://github.com/hercules-ci/hercules-ci-agent/blob/master/hercules-ci-api/src/Hercules/API/Name.hs#L19) ` Account` -> *tech debt*)
 - the `displayName` (such as "Domen Kožar")

Reply:
The pushing account seems to be missing. I think we're trying to answer three questions the user may have:

 1. "who made the last change to this set of files?" - the commit author
 2. "who or what initiated this Job?" - the account that pushed
 3. "who decided that the the set of files is appropriate for <ref> in the customer's workflow?" - the account that pushed

(1) is covered.

(3) and (2) can be covered by tracking the `ref` and pushing account as much as possible, but this may not be feasible in the multi-argument case where we'll have do a lookup by git ref, which will not include the pushing account (unless we're lucky to find some API, but we shouldn't require every git host to provide that info). So in the `GitSource` type, the pushing account should be optional.

(2) is somewhat covered by the above, but it being optional is annoying for the Job frontend, so what we can do is answer (2) directly with a `startedBy` (or similar) attribute on `Job`. This won't be optional and can be extended easily in the future to include manual reruns etc. The backend should have no problem finding a value for this.

Reply:
- `committerName -> committerAuthor` to denote it's without the email?

1. Covered
2. Out of scope - can be added once there's more than one way to trigger a job
3. Out of scope - will be covered once we need the pusher